araig_test_runners/scripts/test_4_static_obstacle

```diff
@@ -0,0 +1,136 @@
+        param_list = extend_param_list, \
+        rate = rate)
+
+    def main(self):
+        if self.pre_state == None and self.current_state == State.wait:
+            rospy.loginfo(rospy.get_name() + ": Waiting to start...")
+            self.pre_state = State.wait
+
+         # start test, start robot (send goal)
+        if self.current_state == State.wait and self.getSafeFlag("start_test"):
+            self.startRecordingAndWait(3)
+            rospy.loginfo(rospy.get_name() + ": Starting robot.")
+            self._publishers["start_robot"].publish(self.buildNewBoolStamped(True))
+            self.current_state = State.start_test
+            self.pre_state = State.start_test
+
+        # robot in collision zoom
+        if self.current_state == State.start_test and self.getSafeFlag("robot_in_collision"):
+            rospy.loginfo(rospy.get_name() + ": Robot reached collision zoom , ending test in failure!")
+            self._publishers["test_failed"].publish(self.buildNewBoolStamped(True))
+            self._publishers["test_completed"].publish(self.buildNewBoolStamped(True))
+            self.current_state = State.complete_test
+
+        # If interrupted
+        if self.current_state == State.start_test and self.getSafeFlag("interrupt_test"):
+            rospy.logwarn(rospy.get_name() + ": Interrupted!!")
+            self._publishers["test_failed"].publish(self.buildNewBoolStamped(True))
+            self._publishers["test_completed"].publish(self.buildNewBoolStamped(True))
+            # !!! ui should publish interrupt_test to False after pub True, interrupt_test signal is event 
+            self._publishers["interrupt_test"].publish(self.buildNewBoolStamped(False))
+            self.current_state = State.complete_test
+
+        # If robot has stopped (because it found an obstacle), but maybe robot will still move
+        if self.current_state == State.start_test and self.getSafeFlag("goal"):
+            rospy.loginfo(rospy.get_name() + ": Robot has reached goal and stop, will check if robot move again in {}s".format(self.config_param['timeout_check_goal']))
+            self.current_state = State.check_condition
+        
+        # Check if robot stop and reach goal again in 2s
+        if self.pre_state == State.start_test and  \
+            self.current_state == State.check_condition:
+            if self.getSafeFlag("goal"):
+                if self.checkCondition(self.getSafeFlag("goal", header = True).stamp, \
+                    self.config_param['timeout_check_goal']):
+                    rospy.loginfo(rospy.get_name() + ": Robot has reached goal and stop.")
+                    self.current_state = State.reached_goal
+            else:
+                rospy.loginfo(rospy.get_name() + ": Robot is still moving, waiting for robot reaching goal and stop")
+                self.current_state = State.start_test
+        
+        # Test complete if robot doesn't move any more and reached the goal
```

Comment:
Well if the oscillations are undamped, this is basically an infinite loop :P So lets not try to compensate for it.